lib/Models/WebMapServiceCatalogItem.js

```diff
@@ -1143,6 +1156,40 @@ WebMapServiceCatalogItem.prototype.exportData = function() {
     callWebCoverageService(this);
 };
 
+WebMapServiceCatalogItem.prototype.searchDataForMatchingFeatures = function(searchText, regex) {
+    const that = this;
+    return new Promise(function(resolve, reject) {
```

Comment:
It's almost never necessary to explicitly construct promises, definitely not necessary here.